### ISSUE TITLE
Unable to compile Freeradius with Libressl-2.9.2 on Gentoo x86_64

### ISSUE THREAD

**Issue by Jacekalex** [FreeRADIUS/freeradius-server#2783]
Hi
My OS:
`Linux version 4.19.58-g1 (root@box) (gcc version 8.3.0 (Gentoo Hardened 8.3.0-r1 p1.1)) #1 SMP PREEMPT Thu Jul 11 13:51:40 CEST 2019

EDIT:
net-dialup/freeradius-3.0.18-r1  solved this problem. ;)
`
Compiler & lib:
```
dev-libs/libressl-2.9.2 abi_x86_32 abi_x86_64 asm static-libs
sys-devel/gcc-8.3.0-r1 cxx fortran hardened multilib nls nptl openmp pie sanitize ssp vtv
sys-libs/glibc-2.29-r2 caps multiarch multilib selinux ssp

```

Error:
```
mkdir -p build/objs/src/modules/rlm_eap/types/rlm_eap_fast/
true CC src/modules/rlm_eap/types/rlm_eap_fast/rlm_eap_fast.c
build/make/jlibtool --mode=compile --shared x86_64-pc-linux-gnu-gcc -o build/objs/src/modules/rlm_eap/types/rlm_eap_fast/rlm_eap_fast.lo -c -MD -march=native -O2 -pipe -I. -Isrc -include src/freeradius-devel/autoconf.h -include src/freeradius-devel/build.h -include src/freeradius-devel/features.h -include src/freeradius-devel/radpaths.h -fno-strict-aliasing -Wno-date-time -march=native -O2 -pipe -Wall -std=c99 -D_GNU_SOURCE -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -DOPENSSL_NO_KRB5 -DNDEBUG -DIS_MODULE=1 -Isrc/modules/rlm_eap -Isrc/modules/rlm_eap/libeap src/modules/rlm_eap/types/rlm_eap_fast/rlm_eap_fast.c
Executing: /bin/sh -c x86_64-pc-linux-gnu-gcc -c -MD -march=native -O2 -pipe -I. -Isrc -include src/freeradius-devel/autoconf.h -include src/freeradius-devel/build.h -include src/freeradius-devel/features.h -include src/freeradius-devel/radpaths.h -fno-strict-aliasing -Wno-date-time -march=native -O2 -pipe -Wall -std=c99 -D_GNU_SOURCE -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -DOPENSSL_NO_KRB5 -DNDEBUG -DIS_MODULE=1 -Isrc/modules/rlm_eap -Isrc/modules/rlm_eap/libeap src/modules/rlm_eap/types/rlm_eap_fast/rlm_eap_fast.c -fPIC -o build/objs/src/modules/rlm_eap/types/rlm_eap_fast/rlm_eap_fast.o  
Executing: touch build/objs/src/modules/rlm_eap/types/rlm_eap_fast/rlm_eap_fast.lo 
mkdir -p build/make/src/src/modules/rlm_eap/types/rlm_eap_fast/
mkdir -p build/objs/src/modules/rlm_eap/types/rlm_eap_fast/
sed  -e 's/#.*//' -e 's,^/var/tmp/portage/net-dialup/freeradius-3.0.17/work/freeradius-server-3.0.17,${top_srcdir},' -e 's, /var/tmp/portage/net-dialup/freeradius-3.0.17/work/freeradius-server-3.0.17, ${top_srcdir},' -e 's,^build,${BUILD_DIR},' -e 's, build/make/include/[^ :]*,,' -e 's, build, ${BUILD_DIR},' -e 's, /[^: ]*,,g' -e 's,^ *[^:]* *: *$,,' -e '/: </ d' -e 's/\.o: /.${OBJ_EXT}: /' -e '/^ *\\$/ d' < build/objs/src/modules/rlm_eap/types/rlm_eap_fast/rlm_eap_fast.d | sed -e '$!N; /^\(.*\)\n\1$/!P; D' >  build/make/src/src/modules/rlm_eap/types/rlm_eap_fast/rlm_eap_fast.mk
sed -e 's/#.*//' -e 's, build/make/include/[^ :]*,,' -e 's, /[^: ]*,,g' -e 's,^ *[^:]* *: *$,,' -e '/: </ d' -e 's/^[^:]*: *//' -e 's/ *\\$//' -e 's/$/ :/' < build/objs/src/modules/rlm_eap/types/rlm_eap_fast/rlm_eap_fast.d | sed -e '$!N; /^\(.*\)\n\1$/!P; D' >> build/make/src/src/modules/rlm_eap/types/rlm_eap_fast/rlm_eap_fast.mk
mkdir -p build/objs/src/modules/rlm_eap/types/rlm_eap_fast/
true CC src/modules/rlm_eap/types/rlm_eap_fast/eap_fast.c
build/make/jlibtool --mode=compile --shared x86_64-pc-linux-gnu-gcc -o build/objs/src/modules/rlm_eap/types/rlm_eap_fast/eap_fast.lo -c -MD -march=native -O2 -pipe -I. -Isrc -include src/freeradius-devel/autoconf.h -include src/freeradius-devel/build.h -include src/freeradius-devel/features.h -include src/freeradius-devel/radpaths.h -fno-strict-aliasing -Wno-date-time -march=native -O2 -pipe -Wall -std=c99 -D_GNU_SOURCE -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -DOPENSSL_NO_KRB5 -DNDEBUG -DIS_MODULE=1 -Isrc/modules/rlm_eap -Isrc/modules/rlm_eap/libeap src/modules/rlm_eap/types/rlm_eap_fast/eap_fast.c
Executing: /bin/sh -c x86_64-pc-linux-gnu-gcc -c -MD -march=native -O2 -pipe -I. -Isrc -include src/freeradius-devel/autoconf.h -include src/freeradius-devel/build.h -include src/freeradius-devel/features.h -include src/freeradius-devel/radpaths.h -fno-strict-aliasing -Wno-date-time -march=native -O2 -pipe -Wall -std=c99 -D_GNU_SOURCE -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS -DOPENSSL_NO_KRB5 -DNDEBUG -DIS_MODULE=1 -Isrc/modules/rlm_eap -Isrc/modules/rlm_eap/libeap src/modules/rlm_eap/types/rlm_eap_fast/eap_fast.c -fPIC -o build/objs/src/modules/rlm_eap/types/rlm_eap_fast/eap_fast.o  
src/modules/rlm_eap/types/rlm_eap_fast/eap_fast.c: In function ‘openssl_get_keyblock_size’:
src/modules/rlm_eap/types/rlm_eap_fast/eap_fast.c:59:20: error: ‘struct ssl3_state_st’ has no member named ‘tmp’
   md_size = ssl->s3->tmp.new_mac_secret_size;
                    ^~
make: *** [scripts/boiler.mk:635: build/objs/src/modules/rlm_eap/types/rlm_eap_fast/eap_fast.lo] Error 1
 * ERROR: net-dialup/freeradius-3.0.17::gentoo failed (compile phase):
 *   emake failed
 * 
```
 Build log:
https://pastebin.com/raw/gNweeaKY

Cheers

**Comment by arr2036** [comment]
Can you try with 3.0.19, there have been various OpenSSL build fixes since 3.0.17

**Comment by Jacekalex** [comment]
Problem Fixed:
`net-dialup/freeradius-3.0.18-r1 ldap libressl mysql odbc pam pcap python_single_target_python3_6 python_targets_python2_7 python_targets_python3_6 readline ssl`

This version works fine. :)

Thanks
